docs/fastdds/xtypes/discovery_matching.rst

```diff
@@ -0,0 +1,102 @@
+     *eProsima Fast DDS* evaluates whether all required types are already known.
+     In cases where some types are missing, an automatic process to discover the necessary types is initiated.
+     Utilizing |TypeObject-api| to communicate with other participants and discover the topic types asynchronously.
+     Matching of the discovered participant is deferred until all necessary types are known.
+
+2. **Fallback Mechanism**: In scenarios where the discovered participant has not transmitted |TypeInformation-api|,
+     *eProsima Fast DDS* resorts to a fallback mechanism.
+     It utilizes the type name to verify if the type is already known within the system.
+     If the type name is recognized, matching occurs.
+
+.. _xtypes_discovery_typeinformation:
+
+TypeInformation
+---------------
+
+|TypeInformation-api| is a fundamental component of the *XTypes 1.3* specification, facilitating the efficient exchange
+of essential data type information between DDS Endpoints.
+Instead of transmitting the entire |TypeObject|, Endpoints share a compact representation containing the |TypeIdV1-api|
+of the respective data type.
+The exchanged |TypeIdV1-api| serve as unique identifiers for the associated data types, allowing Endpoints to verify
+compatibility and also facilitate subsequent retrieval of the necessary |TypeObject-api| for the data types needed,
+enhancing interoperability without burdening non-interested Endpoints with unnecessary data.
+
+This approach significantly reduces network overhead, especially when communicating with Endpoints that may
+not require the complete type definition.
+
+The contents of TypeInformation include:
+
+* |TypeIdV1-api|: This is the core component of TypeInformation,
+  representing the unique identifier of the data type associated with the Endpoint.
+  It enables Endpoints to identify and verify the compatibility of data types during communication.
+
+* Dependency Information: Optionally, TypeInformation may include dependency information,
+  indicating other data types on which the current data type depends.
+  This information aids Endpoints in understanding the data type hierarchy and its dependencies,
+  facilitating comprehensive compatibility checks. The decision regarding which types to include in the
+  dependency information of TypeInformation is left to the implementation.
+
+|TypeInformation-api| is described in the annexes of `DDS-XTypes V1.3`_ document,
+please refer to this document for details.
+
+.. _xtypes_discovery_typeobject:
+
+TypeObject
+----------
+
+The *XTypes* type system uniquely identifies any possible type with a TypeIdentifier.
+For simple types such as primitive types, strings, or certain sequences of primitives,
+the TypeIdentifier fully describes the type.
+However, for more complex types, the TypeIdentifier only serves to identify the type,
+and its complete description requires a TypeObject.
```

Comment:
```suggestion
The *XTypes* type system uniquely identifies each any possible type with a TypeIdentifier.
For simple types, such as primitive types, strings, or certain sequences of primitives,
the TypeIdentifier fully describes the type.
For more complex types, however, the TypeIdentifier only serves to identify the type,
and its full description requires a TypeObject.
```

Reply:
fixed in [e38d0d8](https://github.com/eProsima/Fast-DDS-docs/pull/730/commits/e38d0d856970dc4cdf6d7a04f3490344f7c903b1)